sdks/python/container/boot.go

```diff
@@ -210,15 +224,29 @@ func main() {
 	wg.Add(len(workerIds))
 	for _, workerId := range workerIds {
 		go func(workerId string) {
+			defer wg.Done()
 			log.Printf("Executing: python %v", strings.Join(args, " "))
-			log.Fatalf("Python exited: %v", execx.ExecuteEnv(map[string]string{"WORKER_ID": workerId}, "python", args...))
-			wg.Done()
+			log.Printf("Python exited: %v", execx.ExecuteEnv(map[string]string{"WORKER_ID": workerId}, "python", args...))
 		}(workerId)
 	}
 	wg.Wait()
 }
 
-// setup wheel specs according to installed python version
+// setupVenv initialize a local Python venv and set the corresponding env variables
+func setupVenv(dir string) error {
+	log.Printf("Initializing temporary Python venv ...")
+	if _, err := os.Stat(filepath.Join(dir, "pyenv.cfg")); os.IsNotExist(err) {
```

Comment:
I guess I'm curious. If there is an error other than the IsNotExist error, could anything go wrong?

Reply:
Not sure what you mean. The call to `python -m venv` could go wrong, I guess.

Reply:
So the way I read this code is,
if os.Stat returns an error then only take any action on that error if os.IsNotExist, other errors will silently continue.

If that's what we want then this is probably fine.

Reply:
The point here is to avoid initializing the venv twice. If the file exists, there is nothing left to do.

Reply:
What is the reason to reuse the venv exactly? When do we want to do that as opposed to creating a new venv from scratch?

Reply:
I suspect we would want that if we want to launch multiple processes from the same environment.

Reply:
Same reason the "execute only once" stuff was added: efficiency and thread safety. The latter isn't a problem anymore when we use separate venvs, since we're not installing to the same location anymore. Efficiency may still be an issue, but as written above, I don't think we can get away with reusing the venvs, because the lifecycle of a job is totally underspecified and the worker pool knows nothing about whether it's safe to clean up now.